packages/alert/src/Alert.tsx

```diff
@@ -69,9 +69,34 @@ export default function Alert({
   variant = 'success',
   ...other
 }: AlertProps): React.ReactElement<AlertProps> {
+  const [show, setShow] = useState(true)
   const Icon = icons[variant]
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout
+    if (type === 'toast') {
+      timer = setTimeout(hideAlert, 5 * 1000)
+    }
+
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, [type])
+
+  /* istanbul ignore next */
+  const hideAlert = () => {
+    setShow(false)
+    onClose?.()
+  }
+
   return (
-    <AlertWrapper role="alert" variant={variant} type={type} {...other}>
+    <AlertWrapper
+      role="alert"
+      variant={variant}
+      type={type}
+      className={show ? 'show' : 'hide'}
```

Comment:
How can a consumer dev programatically show a `ToastAlert`?

Reply:
What's a common pattern used by other libraries?

Reply:
This is how it is done in MUI when `Alert`s can be stacked: https://mui.com/material-ui/react-snackbar/#notistack

Reply:
Got it. It might seem it's better than the state is controlled from outside, and what this component do is provide a callback so that the client can change the state.

Something like this:

```
function MyComponent() {
  const [isAlertOpen, setAlertOpen] = useState(true)

  return (
    <>
      <p>Lorem ipsum</p>
      {<Alert isOpen={isAlertOpen} onClose={() => setAlertOpen(false)} delay={5000} />}
    </>
  )
}
```

What changes from the current implementation is that there is no local state inside the alert. What do you think @ryanjdiercks @huhgawz?

Reply:
I agree about controlling the `Alert` open state from outside.

IMHO, the `type="toast"` feels unnecessary. 

I think an additional component should be implemented not only for controlling the `Alert` open state but for managing the stack functionality.